pkg/controller/configmap_ca_injector/controller.go

```diff
@@ -0,0 +1,177 @@
+		},
+	}
+	// The events fire for changes/creation of the trusted-ca-bundle and any configmaps with the
+	//"config.openshift.io/inject-trusted-cabundle" annotation
+	pred := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.MetaNew.GetAnnotations()[names.TRUST_BUNDLE_CONFIGMAP_ANNOTATION] == "true" ||
+				(e.MetaNew.GetName() == names.TRUST_BUNDLE_CONFIGMAP && e.MetaNew.GetNamespace() == names.TRUST_BUNDLE_CONFIGMAP_NS)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return e.Meta.GetAnnotations()[names.TRUST_BUNDLE_CONFIGMAP_ANNOTATION] == "true" ||
+				(e.Meta.GetName() == names.TRUST_BUNDLE_CONFIGMAP && e.Meta.GetNamespace() == names.TRUST_BUNDLE_CONFIGMAP_NS)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return e.Meta.GetAnnotations()[names.TRUST_BUNDLE_CONFIGMAP_ANNOTATION] == "true" ||
+				(e.Meta.GetName() == names.TRUST_BUNDLE_CONFIGMAP && e.Meta.GetNamespace() == names.TRUST_BUNDLE_CONFIGMAP_NS)
+		},
+	}
+
+	err = c.Watch(&source.Kind{Type: cfpMap}, &handler.EnqueueRequestForObject{}, pred)
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileConfigMapInjector{}
+
+type ReconcileConfigMapInjector struct {
+	client client.Client
+	scheme *runtime.Scheme
+	status *statusmanager.StatusManager
+}
+
+// Reconcile expectes requests to refers to configmaps of two different types.
+// 1. a configmap named trusted-ca-bundle in namespace openshift-config-managed and will ensure that all configmaps with the annotation
+// config.openshift.io/inject-trusted-cabundle = true have the certificate information stored in trusted-ca-bundle's ca-bundle.crt entry.
+// 2. a configmap with the annotation config.openshift.io/inject-trusted-cabundle = true and will insure that it contains the ca-bundle.crt
+// entry in the configmap named trusted-ca-bundle in namespace openshift-config-managed.
+func (r *ReconcileConfigMapInjector) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.Printf("Reconciling configmap from  %s/%s\n", request.Name, request.Namespace)
+
+	trustedCAbundleConfigMap := &corev1.ConfigMap{}
+	trustedCAbundleConfigMapName := types.NamespacedName{
+		Namespace: names.TRUST_BUNDLE_CONFIGMAP_NS,
+		Name:      names.TRUST_BUNDLE_CONFIGMAP,
+	}
+	err := r.client.Get(context.TODO(), trustedCAbundleConfigMapName, trustedCAbundleConfigMap)
+	if err != nil {
+		log.Println(err)
+		return reconcile.Result{}, err
+	}
+	trustedCAbundleData, err := r.validateTrustBundleConfigMap(trustedCAbundleConfigMap)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	//build a list of configMaps
+	configMapsToChange := []corev1.ConfigMap{}
+
+	//The trusted-ca-bundle changed
+	if request.Name == names.TRUST_BUNDLE_CONFIGMAP && request.Namespace == names.TRUST_BUNDLE_CONFIGMAP_NS {
+
+		configMapList := &corev1.ConfigMapList{}
+		err = r.client.List(context.TODO(), &client.ListOptions{}, configMapList)
```

Comment:
this is going to list every configmap in the cluster, right?  i really think this needs to move to a listwatcher so we have a cache, otherwise this is going to put a lot of load on the apiserver w/ every reconcile loop.

Reply:
@bparees AFAICT there's no way to get an actual KubeClient in this code (controller-runtime doesn't seem to provide it). But @JacobTanenbaum I think what you can do is to use the Predicate create/delete/update functions for your cache handling. So for example you'd have a `configMapCache map[types.NamespacedName]ConfigMap` as your cache and whenever your Predicate's Create is called if the event.Object passes your test, add it to the cache. Likewise whenever your Predicate's Delete is called remove that object from the cache. Update is a bit tricker because if the object doesn't  pass the test, you need to delete it from the cache, but if it passes the test you add it to the cache.

Then in your Reconcile you can just range over the cache map instead of re-grabbing stuff.

@bparees that said, doesn't all this come from a cache inside the shared indexer anyway? If true, why would we double-cache things?

Reply:
yeah i didn't realize you guys are using the operator-sdk which gives you a caching client.  I thought it was just a normal client-go client.  This is fine, if indeed the client is backed by a local cache that's being updated via watch events.

Reply:
Regardless of how this is currently implemented, the operator-sdk team is receptive to the idea of a label-scoped watch - https://github.com/operator-framework/operator-sdk/issues/1808

Reply:
i've gotten confirmation that proceeding w/ using a label is fine.

Reply:
fixed